hips/draw/simple.py

```diff
@@ -1,76 +1,89 @@
 from ..tiles import HipsSurveyProperties, HipsTile, HipsTileMeta
-from ..utils import WCSGeometry, compute_healpix_pixel_indices
+from ..utils import WCSGeometry, compute_healpix_pixel_indices, get_hips_order_for_resolution
+
 
 __all__ = [
-    'draw_sky_image',
     'make_sky_image',
     'SimpleTilePainter',
-    'compute_matching_hips_order'
+
 ]
 
 __doctest_skip__ = [
-    'compute_matching_hips_order',
+    'SimpleTilePainter'
 ]
 
 
-# TODO: Fix type annotation issue
-def draw_sky_image(geometry: WCSGeometry, tiles: Generator[HipsTile, Any, Any],
-                   hips_survey: HipsSurveyProperties, tile_format: str) -> np.ndarray:
-    """Draw sky image using the simple and quick method.
+class SimpleTilePainter:
+    """Paint HiPS tiles onto an all-sky image using a simple projective
+    transformation method. The algorithm implemented is described
+    here: :ref:`drawing_algo`.
 
     Parameters
     ----------
     geometry : `~hips.utils.WCSGeometry`
         An object of WCSGeometry
-    tiles : List[HipsTile]
-        A list of HipsTile
     hips_survey : `~hips.HipsSurveyProperties`
         HiPS survey properties
     tile_format : `str`
         Format of HiPS tile
 
-    Returns
-    -------
-    np.ndarray
-        Returns a numpy array containing all HiPS tiles projected onto it
-    """
-    if tile_format == 'jpg':
-        shape = (geometry.shape.height, geometry.shape.width, 3)
-    elif tile_format == 'png':
-        shape = (geometry.shape.height, geometry.shape.width, 4)
-    else:
-        shape = (geometry.shape.height, geometry.shape.width)
-    image = np.zeros(shape)
-    for tile in tiles:
-        painter = SimpleTilePainter(geometry, hips_survey, tile)
-        image += painter.warp_image()
-    return image
-
-
-class SimpleTilePainter:
-    """Paint a single tile using a simple projective transformation method.
-    The algorithm implemented is described here: :ref:`drawing_algo`.
-
-    Parameters
-    ----------
-    geometry : `~hips.utils.WCSGeometry`
-        An object of WCSGeometry
-    hips_survey : `~hips.HipsSurveyProperties`
-        HiPS survey properties
-    tile : `HipsTile`
-       An object of HipsTile
+    Examples
+    --------
+    >>> from hips.utils import WCSGeometry
+    >>> from hips.draw import SimpleTilePainter
+    >>> from hips.tiles import HipsSurveyProperties
+    >>> from astropy.coordinates import SkyCoord
+    >>> url = 'http://alasky.unistra.fr/DSS/DSS2Merged/properties'
+    >>> hips_survey = HipsSurveyProperties.fetch(url)
+    >>> geometry = WCSGeometry.create_simple(
+    ...     skydir=SkyCoord(0, 0, unit='deg', frame='icrs'),
+    ...     width=2000, height=1000, fov="3 deg",
+    ...     coordsys='icrs', projection='AIT'
+    ... )
+    >>> simple_tile_painter = SimpleTilePainter(geometry, hips_survey, 'fits')
+    >>> simple_tile_painter.hips_order
+    7
+    >>> simple_tile_painter.run()
+    >>> simple_tile_painter.image[0]
```

Comment:
This is not the current output. It would be:
```
array([ 1866.7819988 ,  1865.42884071,  2494.00873715, ...,  1970.00283741,
        1998.08389507,  2045.58599973])
```
But I suggest you just do `painter.shape` instead of `painter.image[0]` here.